File: parent335/pinyougou-interface/src/main/java/cn/itcast/core/service/user/UserService.java

package cn.itcast.core.service.user;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.user.User;
import cn.itcast.core.pojo.user.UserHot;

public interface UserService {

    /**
     * 用户获取短信验证码
     *
     * @param phone
     */
    public void sendCode(String phone);

    /**
     * 用户注册
     *
     * @param user
     * @param smscode
     */
    public void add(User user, String smscode);

    void deleteOne(Long id);

    PageResult search(Integer page, Integer rows, User user);

    void deleteMany(Long[] ids);

    void addToRedis(String username);

    void updatelastlogintime(String username);

    Integer getcurrentlogincount();

    void savecurrentlogin();

    UserHot findUserHotByDate(String date);

    /**
     * 用户 我的订单查询
     * @param page
     * @param rows
     * @param userId
     * @return
     */
    public PageResult findAll(Integer page, Integer rows, String userId);
}

File: parent335/pinyougou-web-manager/src/main/webapp/js/service/userService.js
//服务层
app.service('userService', function ($http) {

    //删除单个
    this.deleteOne = function (id) {
        return $http.get('../user/deleteOne.do?id=' + id);
    }


    //删除多个
    this.deleteMany = function (ids) {
        return $http.get('../user/deleteMany.do?ids=' + ids);
    }
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../user/search.do?page=' + page + "&rows=" + rows, searchEntity);
    }

    this.excel = function (startTime,endTime,consumesType,conPaymentStatus) {
        return $http.get("../user/exportExcel.do?startTime=" + startTime
            + "&endTime=" + endTime + "&consumesType="
            + consumesType + "&conPaymentStatus=" + conPaymentStatus
        );
    }
});
